numba/tests/test_extending.py

```diff
@@ -1374,6 +1374,26 @@ def codegen(context, builder, signature, args):
         second = pickle.loads(pickled)
         self.assertIs(rebuilt._defn, second._defn)
 
+    def test_docstring(self):
+
+        @intrinsic
+        def void_func(typingctx, a: int):
+            """void_func docstring"""
+            sig = types.void(types.int32)
+
+            def codegen(context, builder, signature, args):
+                pass  # do nothing, return None, should be turned into
+                # dummy value
+
+            return sig, codegen
+
+        self.assertEqual("numba.tests.test_extending", void_func.__module__)
+        self.assertEqual("void_func", void_func.__name__)
+        self.assertEqual("TestIntrinsic.test_docstring.<locals>.void_func",
+                         void_func.__qualname__)
+        self.assertDictEqual({'a': int}, void_func.__annotations__)
```

Comment:
I wasn't aware these were added. I'll need to try in my next PR.

Reply:
`functools.update_wrapper` updates the following fields: `__module__`, `__name__`, `__qualname__`, `__annotations__` and `__doc__` ([ref](https://docs.python.org/3/library/functools.html#functools.update_wrapper))

Reply:
oops, I mean `assertDictEqual`, not anything particular in your PR.  (I was rushing to finish up the review before starting a meeting and didn't realize the confusing comment.)

Reply:
Oh, ok 😄